### PR TITLE
Issue412

### LINKED ISSUE #412: Glob pattern in trigger.watchPattern doesn't work...
**Describe the bug**
When trying to use a glob pattern like '/aiinput/**/Dog*.jpg' in trigger.watchPattern exeption in verifyTriggerWatchLocation method is raised ( Unable to read contents of watch folder ${trigger.watchPattern} for trigger ${trigger.name}. Check and make sure the image folder is mounted properly. ${e})

**Log entries**
Unable to read contents of watch folder ${trigger.watchPattern} for trigger ${trigger.name}. Check and make sure the image folder is mounted properly. ${e}

**Installation details**
 - OS: Unraid
 - Docker setup UnRAID

**Additional context**

### PR DESCRIPTION
**Pull request by salcio**
# Fixes #412 

## Description of changes
Modified verifyTriggerWatchLocations to use glob.sync(trigger.watchPattern) instead of fs.readdirsync()
## Checklist

If your change touches anything under src or the README.md file
these items must be done:

- [ x] User-facing change description added to unreleased section of CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 
+## Unreleased
+
+- Support for glob patterns in trigger watchPatterns. Resolves [issue 412](https://github.com/danecreekphotography/node-deepstackai-trigger/issues/412)
```
Comment (danecreekphotography) [inline CHANGELOG.md:5]:
```suggestion
- Support for glob patterns in trigger watchPatterns. Resolves [issue 412](https://github.com/danecreekphotography/node-deepstackai-trigger/issues/412).
```

File: .gitignore [hunk 2]
```diff
@@ -108,4 +108,4 @@ dist
 
 # sample configuration files
 triggers.json
-settings.json
\ No newline at end of file
+settings.json
```
Comment (danecreekphotography) [inline .gitignore:111]: This still seems to be in the PR and shouldn't be.

**Review by danecreekphotography — CHANGES_REQUESTED** [review]
Thanks for the PR! A couple of general comments:

1. There's a build failure that needs to be resolved
2. Does this work with and without a glob pattern in watchFolder? I assume yes since you only changed dog, and not cat, detector in the sample.json file.

File: .gitignore (hunk 2 above)
Comment (salcio) [reply .gitignore:111]: I'll remove it then.

**Comment by salcio** [comment]
> Thanks for the PR! A couple of general comments:
> 
> 1. There's a build failure that needs to be resolved
> 2. Does this work with and without a glob pattern in watchFolder? I assume yes since you only changed dog, and not cat, detector in the sample.json file.

yes it works with and without the glob in pattern. 

I'm working on the build failure (something to do with package-lock).

**Review by danecreekphotography — CHANGES_REQUESTED** [review]
Thanks for fixing the build failure! Looks like there are two build warnings now that need cleaning up in triggermanager.ts due to imports that are no longer used.

**Comment by danecreekphotography** [comment]
Great thanks! I'll take another quick look at this tonight then merge it and produce a `dev` docker image for testing. Will run it locally a bit and ask you to try the dev build as well. If that all works out then I'll publish it as a release.

**Comment by salcio** [comment]
> Thanks for fixing the build failure! Looks like there are two build warnings now that need cleaning up in triggermanager.ts due to imports that are no longer used.

Yeah you are right. I haven't seen the warnings. Now fixed. 
You might want to consider settings lint to cause build failure when warnings like that happen - will save time :).

**Comment by danecreekphotography** [comment]
> > Thanks for fixing the build failure! Looks like there are two build warnings now that need cleaning up in triggermanager.ts due to imports that are no longer used.
> 
> Yeah you are right. I haven't seen the warnings. Now fixed.
> You might want to consider settings lint to cause build failure when warnings like that happen - will save time :).

Solid suggestion. Feel free to open a bug and a PR to get that change in 😂 At the very least open an issue to track it so I can get to it next time I'm touching code.

**Comment by salcio** [comment]
> > > Thanks for fixing the build failure! Looks like there are two build warnings now that need cleaning up in triggermanager.ts due to imports that are no longer used.
> > 
> > 
> > Yeah you are right. I haven't seen the warnings. Now fixed.
> > You might want to consider settings lint to cause build failure when warnings like that happen - will save time :).
> 
> Solid suggestion. Feel free to open a bug and a PR to get that change in 😂 At the very least open an issue to track it so I can get to it next time I'm touching code.

Make sense. Issue created #414

**Review by danecreekphotography — CHANGES_REQUESTED** [review]
Code looks good. I'll re-work the text in the release notes when I push the release out.

One requested change: instead of `aiinput/asdf` as the sub-folder please change to a real folder name that matches what's inside. I suggest `aiinput/dogcamera` or similar.

**Comment by salcio** [comment]
> Code looks good. I'll re-work the text in the release notes when I push the release out.
> 
> One requested change: instead of `aiinput/asdf` as the sub-folder please change to a real folder name that matches what's inside. I suggest `aiinput/dogcamera` or similar.

yeah I should have pay more attention before submitting PR. Done some testing and then you see :). 
It's changed now.

**Comment by neileadobe** [comment]
@salcio Heads up it looks like you merged another branch into this one.

Note that if you are planning to open a future pull request to archive the input files I'm unlikely to take it. I don't believe this system should be responsible for managing files it didn't create.

**Comment by salcio** [comment]
> @salcio Heads up it looks like you merged another branch into this one.
> 
> Note that if you are planning to open a future pull request to archive the input files I'm unlikely to take it. I don't believe this system should be responsible for managing files it didn't create.

uuu I didn't want to do that. I wanted to merge the other way. I'll revert that merge.

Makes sense with not touching the files that were created by someone else. I didn't plan to create PR for this as it was more for my setup (ease of adding functionality to existing app rather then creating completely new one)

**Pull request merged**